packages/react-client/src/ReactFlightClient.js

```diff
@@ -2014,8 +2048,19 @@ function parseModelString(
           if (value.length > 2) {
             const debugChannel = response._debugChannel;
             if (debugChannel) {
-              const ref = value.slice(2);
-              debugChannel('R:' + ref); // Release this reference immediately
```

Comment:
We don't have a good way to release unreachable references since we don't know they exist inside a payload until we parse it and when we parse it, we immediately ask for it.

References held by the server aren't released until all lazy references of the whole response are GC:ed.